libr/debug/p/native/linux/linux_debug.c

```diff
@@ -1218,9 +1218,11 @@ RList *linux_desc_list (int pid) {
 		}
 		memcpy (file, path, len);
 		memcpy (file + len, de->d_name, len2 + 1);
-		memset (buf, 0, sizeof(buf));
-		(void)readlink (file, buf, sizeof (buf) - 1);
-		buf[sizeof (buf)-1] = 0;
+		buf[0] = 0;
+		if (readlink (file, buf, sizeof (buf) - 1) == -1) {
+			return NULL;
+		}
+		buf[sizeof (buf) - 1] = 0;
```

Comment:
By the way, we use this pattern so often, it makes sense to make a macro for this. `LAST0(buf)`, something like that, or `LAST0SZ(buf, sz)` in case of the `type *` buf.